R/latent_individual.R

```diff
@@ -15,14 +52,61 @@ epidist_prepare.epidist_latent_individual <- function(data, ...) {
   data[, swindow_upr := stime_upr - stime_lwr]
   data[, delay_central := stime_lwr - ptime_lwr]
   data[, row_id := seq_len(.N)]
-
   if (nrow(data) > 1) {
     data <- data[, id := as.factor(id)]
   }
-
+  validate_latent_individual(data)
```

Comment:
Yes that's correct that it double includes code.

There are reasons to do it this way: it has these checks before the manipulations happen so that the user gets an error from the check not from some `data.table` thing not working. Then the use of the `validate` function was recommended to me at the end of the function, and there isn't a clear way to say "skip the parts that have already been validated".

Open to changing it but not a clear one IMO.

Reply:
And agree it could be better in general to try to convert columns before rejecting them. As well as `as_double` could try `as_integer` etc.

Reply:
Look at `as_forecast` in `scoringutils::` for names passing in

Reply:
Implemented helper function and added issue about type conversion (https://github.com/epinowcast/epidist/issues/146).